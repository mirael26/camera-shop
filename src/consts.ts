export const AppUrl = {
  Main: '/',
  Catalog: '/catalog',
  Page: '/page_',
  Product: '/product',
  Cart: '/cart',
  NotFound: '/404',
  UnknownError: '/unknown_error',
  ServerUnavailable: '/server_unavailable',
} as const;

export const ApiUrl = {
  Promo: '/promo',
  Products: '/cameras',
  Reviews: '/reviews',
  Similar: '/similar',
  Promocode: '/coupons',
  Order: '/orders',
} as const;

export const Modal = {
  AddToCart: 'addToCartModal',
  DeleteFromCart: 'deleteFromCartModal',
  Review: 'reviewModal',
  OrderSuccess: 'orderSuccessModal',
} as const;

export const Tab = {
  Features: 'features',
  Description: 'description',
} as const;

export const Month = {
  0: 'января',
  1: 'февраля',
  2: 'марта',
  3: 'апреля',
  4: 'мая',
  5: 'июня',
  6: 'июля',
  7: 'августа',
  8: 'сентября',
  9: 'октября',
  10: 'ноября',
  11: 'декабря',
} as const;

export const Param = {
  Sort: 'sort',
  Order: 'order',
  Page: 'page',
  Tab: 'tab',
  PriceMin: 'price_min',
  PriceMax: 'price_max',
  Category: 'category',
  Level: 'level',
  Type: 'type',
} as const;

export const SortType = {
  Price: 'price',
  Rating: 'rating',
} as const;

export const SortOrder = {
  Asc: 'asc',
  Desc: 'desc',
} as const;

export const Category = {
  Camera: 'Фотоаппарат',
  VideoCamera: 'Видеокамера',
} as const;

export const Level = {
  Beginner: 'Нулевой',
  Amateur: 'Любительский',
  Professional: 'Профессиональный',
} as const;

export const CameraType = {
  Digital: 'Цифровая',
  Film: 'Плёночная',
  Instant: 'Моментальная',
  Collectible: 'Коллекционная',
} as const;
