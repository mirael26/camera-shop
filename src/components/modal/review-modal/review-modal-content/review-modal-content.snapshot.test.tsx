import { fireEvent, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { AppUrl } from '../../../../consts';
import { renderTestApp } from '../../../../test/helpers/render-test-app';
import { renderWithReduxAndRouter } from '../../../../test/helpers/render-with-redux-and-router';
import ReviewModalContent from './review-modal-content';

jest.mock('../review-form/review-form', () => 'ReviewForm');

test('ReviewModalContent Renders correctly', () => {
  const { asFragment } = renderWithReduxAndRouter(<ReviewModalContent/>);
  expect(asFragment()).toMatchSnapshot();
});
