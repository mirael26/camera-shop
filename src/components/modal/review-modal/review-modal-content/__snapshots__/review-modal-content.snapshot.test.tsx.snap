// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewModalContent Renders correctly 1`] = `
<DocumentFragment>
  <div
    class="modal is-active"
    data-testid="review-modal"
  >
    <div
      class="modal__wrapper"
    >
      <div
        class="modal__overlay"
        data-testid="modal-overlay"
      />
      <reviewform />
    </div>
  </div>
</DocumentFragment>
`;
